src/rosconsole/rosconsole.cpp

```diff
@@ -426,7 +426,12 @@ void Formatter::print(void* logger_handle, ::ros::console::Level level, const ch
   ss << COLOR_NORMAL;
 
   fprintf(f, "%s\n", ss.str().c_str());
-  
+
+  if (f == stderr)
+  {
+    fprintf(stdout, "%s\n", ss.str().c_str());
```

Comment:
Why should the message be printed to `stdout` too if `f == stderr`?

Reply:
As mentioned before, when I use ROS_INFO, the log can be recorded in the node log file, but when I use ROS_ERROR, it is recorded in the rosout.log file and displayed on the screen. It is difficult to search this error log in the huge rosout.log file. Therefore, I'm trying to record ROS_ERROR and ROS_WARN to the node log file.
If there are other ways, then please let me know.

Reply:
I don't think that that this use case alone justifies writing all error messages to two streams. This is a severe change in behavior which I am not comfortably with (without significant rational why that is the only viable option).